tlwatch/auction_backend.py

```diff
@@ -0,0 +1,75 @@
+import functools
+import logging
+import socket
+from typing import Dict, List
+from urllib import parse
+
+import click
+import requests
+
+import bernhard
+from tlwatch import util
+
+logger = logging.getLogger(__name__)
+
+
+def get_latest_block_number(base_url: str) -> int:
+    logger.info(f"Fetch latest block number from '{base_url}'")
+
+    response = requests.get(f"{base_url}/latest-block", timeout=10.0)
+    response.raise_for_status()
+    latest_block_number = response.json().get("blockNumber")
+    return latest_block_number
+
+
+def watch_auction_backend(base_url: str) -> List[Dict]:
+    base_url_parsed = parse.urlparse(base_url)
+    base_url_without_credentials = base_url_parsed._replace(
+        netloc=base_url_parsed.hostname
+    ).geturl()
+
+    try:
+        block_number = get_latest_block_number(base_url)
+        state = "ok"
+        description = f"Retrieved block number: {block_number}"
+
+    except KeyboardInterrupt:
+        raise
+
+    except BaseException as exc:
+        state = "error"
+        description = (
+            f"An error occurred while trying to fetch the latest "
+            f"block number from '{base_url_without_credentials}': {exc}"
+        )
+
+    logger.info(f"Report to Riemann: {state}, {description}")
+
+    return [
+        {
+            "service": base_url_without_credentials,
+            "host": socket.gethostname(),
+            "state": state,
+            "description": description,
+            "ttl": 30,
+        }
```

Comment:
Please report the blocknumber as 'metric' field.

Reply:
So I use `metric` for `state: "ok"` and `description` for `state: "error"`?

Reply:
I have done so. :heavy_check_mark: